var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -121,6 +122,9 @@ class PyNumpy(PythonPackage):
     patch('check_executables4.patch', when='@1.14.0:1.15.4')
     patch('check_executables5.patch', when='@:1.13.3')
 
+    patch('fix_blas_detection.patch', when='@:1.16.99')
+    patch('ibm_fortran.patch')
```

Comment:
Have these patches been submitted upstream? It would be good to get the developers' opinions on these

Reply:
Nope, they need to be released upstream.

```fix_blas_detection.patch``` was made by taking changes introduced in 1.17 and backporting them in order to make blas detection more flexible.

```ibm_fortran.patch``` needs to go upstream.  I think it was originally written without access to actual hardware just reading specs.